utils/s3/client.cc

```diff
@@ -234,26 +235,92 @@ future<> map_s3_client_exception(std::exception_ptr ex) {
+                return repeat_until_value([&gc, &retries, &request, &handler, this]() -> future<std::optional<uint32_t>> {
+                    if (retries <= _retry_strategy->get_max_retries()) {
+                        return gc.http.make_request(request, handler, std::nullopt)
+                                .then([&retries] {
+                                    return make_ready_future<std::optional<uint32_t>>(retries);
+                                })
+                                .handle_exception_type([this, &retries](const aws::aws_exception& ex) mutable -> future<std::optional<uint32_t>> {
+                                    ++retries;
+                                    if (_retry_strategy->should_retry(ex.error(), retries)) {
+                                        s3l.debug("S3 client request failed. Reason: {}. Retry# {}", ex.what(), retries);
+                                        co_await seastar::sleep(std::chrono::milliseconds(_retry_strategy->delay_before_retry(ex.error(), retries)));
+                                    } else {
+                                        if (!ex.error().is_retryable()) {
+                                            s3l.error("S3 client encountered non-retryable error. Reason: {}. Code: {}. Retry# {}", ex.what(),
+                                                    std::to_underlying(ex.error().get_error_type()), retries);
+                                        } else {
+                                            s3l.error("Retries exhausted. Retry# {}", retries);
+                                        }
+                                        co_return coroutine::return_exception(ex);
+                                    }
+                                    co_return std::optional<uint32_t>(std::nullopt);
+                                });
+                    }
+                    return make_exception_future<std::optional<uint32_t>>(std::runtime_error("Retries exhausted"));
+                }).then([](auto) {
+                    return make_ready_future<>();
+                });
+            })
+            .handle_exception([](auto ex) {
+                return map_s3_client_exception(std::move(ex));
+            });
+}
+
+static http::experimental::client::reply_handler make_error_aware_handler(
+        const http::request& req, http::experimental::client::reply_handler&& handler, http::reply::status_type expected) {
+    return [is_mpu_completion{req._method == "POST" && !req.get_query_param("uploadId").empty()}, handler{std::move(handler)}, expected](
+                   const http::reply& rep, input_stream<char>&& in) -> future<> {
+        auto payload = std::move(in);
+        auto error = co_await look_for_errors(rep, payload, is_mpu_completion);
+        if (error.get_error_type() != aws::aws_error_type::OK)
+            co_await coroutine::return_exception(aws::aws_exception(std::move(error)));
```

Comment:
please add braces around this single-line statement for better readability.

Reply:
opps I did it again
will fix

WDYT about flipping the `InsertBraces` to true [here](https://github.com/scylladb/scylladb/blob/a11ede758ed9ae0d251c14769bbdb90a336456a5/.clang-format#L115)